src/lib/auth.ts

```diff
@@ -141,7 +139,7 @@ export const DEVICE_API_KEY_PERMISSIONS = [
 	'resin.image_install.read?device/canAccess()',
 	`resin.image_install.create?device/any(d:${matchesNonFrozenDeviceActor(
 		'd',
-	)}) and installs__image/any(i:i/image__is_part_of__release/any(ipr:ipr/is_part_of__release/any(r:r/belongs_to__application/any(a:a/${ownsNonFrozenDevice} or a/is_public eq true))))`,
+	)}) and installs__image/any(i:i/image__is_part_of__release/any(ipr:ipr/is_part_of__release/any(r:r/belongs_to__application/any(a:a/${ownsDevice} or a/is_public eq true))))`,
```

Comment:
The reasoning being that we already check for is_frozen on the `device/any(d:${matchesNonFrozenDeviceActor('d')})` part.